main/field.c

```diff
@@ -824,30 +824,20 @@ static bool defaultDoesContainAnyChar (const tagEntryInfo *const tag CTAGS_ATTR_
 extern bool  doesFieldHaveTabOrNewlineChar (fieldType type, const tagEntryInfo *tag, int index)
 {
 	fieldObject *fobj = fieldObjects + type;
-	const char *value;
 	bool (* doesContainAnyChar) (const tagEntryInfo *const, const char*, const char*) = fobj->def->doesContainAnyChar;
 
 	Assert (tag);
-	Assert (index == NO_PARSER_FIELD || ((unsigned int)index) < tag->usedParserFields);
+	Assert (index == NO_PARSER_FIELD || (index >= 0 && ((unsigned int)index) < tag->usedParserFields));
 
-	if (doesContainAnyChar == NULL)
-	{
-		if (index == NO_PARSER_FIELD)
-			return false;
-		else
-			doesContainAnyChar = defaultDoesContainAnyChar;
-	}
+	if (index == NO_PARSER_FIELD)
+		return false;
```

Comment:
Your change doesn't consider the combination of `index == NO_PARSER_FIELD` and `doesContainAnyChar != NULL`.